src/request.ts

```diff
@@ -172,16 +171,13 @@ class DatastoreRequest {
    *   const apiResponse = data[1];
    * });
    */
-  allocateIds(key, options, callback) {
-    if (entity.isKeyComplete(key)) {
-      throw new Error('An incomplete key should be provided.');
-    }
+  allocateIds(key: entity.Key, options: AllocateIdsOptions|number): Promise<google.datastore.v1.AllocateIdsResponse>;
+  allocateIds(key: entity.Key, options: AllocateIdsOptions|number, callback: AllocateIdsCallback): void;
+  allocateIds(key: entity.Key, options: AllocateIdsOptions|number, cb?: AllocateIdsCallback): void|Promise<google.datastore.v1.AllocateIdsResponse> {
+    if (entity.isKeyComplete(key)) throw new Error('An incomplete key should be provided.');
```

Comment:
I'm surprised the formatter allows this.  Can you add a newline and brackets to make this easier to read?

Reply:
Done! Should I wrap things like `google.datastore.v1.AllocateIdsResponse` in extended interfaces to make them easier to read?